mesonbuild/backend/backends.py

```diff
@@ -756,17 +770,27 @@ def rpaths_for_non_system_absolute_shared_libraries(self, target: build.BuildTar
                 if os.path.splitext(libpath)[1] not in ['.dll', '.lib', '.so', '.dylib']:
                     continue
 
-                try:
-                    commonpath = os.path.commonpath((libdir, srcdir))
-                except ValueError: # when paths are on different drives on Windows
-                    commonpath = ''
+                if mesonlib.is_windows() and libpath.endswith('.lib'):
+
+                    if isinstance(dep, dependencies.PkgConfigDependency):
+                        # If by chance pkg-config knows the bin dir...
+                        bindir = dep.get_pkgconfig_variable('bindir', [], default='')
+                        if bindir:
+                            add_path(bindir)
+                            continue
+
+                    # If the dll is not in the same dir, try to replace 'lib' with 'bin' in the path.
+                    # This is not 100% proof, since the bin dir could be elsewhere or with a different
+                    # name, and the DLL could have a different name as well, but this is better than nothing.
+                    p = Path(libpath)
+                    if not p.with_suffix('.dll').exists():
```

Comment:
Why not make p = Path(libpath).with_suffix('dll')? Every time you use it you either replace the suffix or get the dirname...

Reply:
good catch!